pkg/cloudprovider/aws/cloudprovider.go

```diff
@@ -120,20 +120,26 @@ func (c *CloudProvider) Delete(ctx context.Context, node *v1.Node) error {
 }
 
 func (c *CloudProvider) GetRequirements(ctx context.Context, provider *v1alpha5.Provider) (scheduling.Requirements, error) {
+	instanceTypes, err := c.GetInstanceTypes(ctx)
+	if err != nil {
+		return nil, fmt.Errorf("getting instance types, %w", err)
+	}
+	requirements := cloudprovider.InstanceTypeRequirements(instanceTypes)
+
 	awsprovider, err := v1alpha1.Deserialize(provider)
 	if err != nil {
-		return scheduling.NewRequirements(), apis.ErrGeneric(err.Error())
+		return nil, apis.ErrGeneric(err.Error())
 	}
 	// Constrain AZs from subnets
 	subnets, err := c.subnetProvider.Get(ctx, awsprovider)
 	if err != nil {
-		return scheduling.NewRequirements(), err
```

Comment:
Not sure I understand why we were returning `scheduling.NewRequirements()` earlier. `nil` makes more sense to me.

Reply:
Because it used to be a concrete type. Now it's a map which is nillable.